### PR TITLE
refactor: migrated help screen to compose

### LINKED ISSUE #2447: refactor : Migration of Help fragment from Xml to Compose view
**Summary:**

Compose offers a more concise and expressive syntax for UI development compared to XML.

### PR DESCRIPTION
**Pull request by AvneetSingh2001**
Fixes #2447 

Screen Recording (Light Mode):
https://github.com/openMF/mifos-mobile/assets/77102514/392d4c49-7b60-49ae-87b3-e284c2027741

Screenshot (Dark Mode) 
screenshot_20240123_120801_mifos_mobile_720

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by therajanmaurya** [comment]
Thanks for the PR! Let me review

File: app/src/main/java/org/mifos/mobile/ui/help/HelpScreen.kt [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        Text(
+            text = stringResource(id = R.string.faq),
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 16.dp, end = 16.dp, top = 12.dp, bottom = 8.dp),
+            style = MaterialTheme.typography.titleMedium,
+            color = if (isSystemInDarkTheme()) Color.White else Color.Black
+        )
+
+        if (!faqArrayList.isNullOrEmpty() && faqArrayList.isNotEmpty()) {
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobile/ui/help/HelpScreen.kt]: `!faqArrayList.isNullOrEmpty()` is enough, don't ned to add `faqArrayList.isNotEmpty()`

File: app/src/main/java/org/mifos/mobile/ui/help/HelpTopBar.kt [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mifos.mobile.core.ui.component.MifosSearchTextField
+
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun HelpTopBar(
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobile/ui/help/HelpTopBar.kt]: Add new module called `ui` and make this generic and move there.

File: core/src/main/java/org/mifos/mobile/core/ui/component/EmptyDataView.kt [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+@Composable
+fun EmptyDataView(
```
Comment (therajanmaurya) [inline core/src/main/java/org/mifos/mobile/core/ui/component/EmptyDataView.kt]: Move in new module `ui` as suggested.

File: core/src/main/java/org/mifos/mobile/core/ui/component/MifosSearchTextField.kt [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.TextFieldValue
+
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun MifosSearchTextField(
```
Comment (therajanmaurya) [inline core/src/main/java/org/mifos/mobile/core/ui/component/MifosSearchTextField.kt]: move it to `ui` module as suggested.

File: core/src/main/java/org/mifos/mobile/core/ui/component/MifosTextButtonWithTopDrawable.kt [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+
+
+@Composable
+fun MifosTextButtonWithTopDrawable(
```
Comment (therajanmaurya) [inline core/src/main/java/org/mifos/mobile/core/ui/component/MifosTextButtonWithTopDrawable.kt]: move it to ui module as suggested.

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

**Comment by therajanmaurya** [comment]
@AvneetSingh2001 Please make suggested changes.

File: app/src/main/java/org/mifos/mobile/ui/help/HelpTopBar.kt (hunk 2 above)
Comment (PratyushSingh07) [reply app/src/main/java/org/mifos/mobile/ui/help/HelpTopBar.kt]: Hey @AvneetSingh2001 no need to create a new module. You can just rename `core` to `ui`

**Comment by PratyushSingh07** [comment]
Hey @AvneetSingh2001 I went through the recording that you have attached. There are a couple of things that I would like to point out:
- text size of the query in search bar looks big
- after you change the orientation (landscape to portrait and vice versa) I saw that the rest of the questionnaire lost their visibility. Is it actually a bug or is it just the screen recording? I am attaching the screenshot below
https://github.com/openMF/mifos-mobile/assets/90026952/0233a512-4b96-4ea0-9623-9f462b375511

**Comment by AvneetSingh2001** [comment]
@therajanmaurya Thanks for reviewing it, I will make the necessary changes.

@PratyushSingh07, thanks for mentioning the bugs, Will make the search bar text smaller.  Additionally for the second point that you have mentioned, Its not a bug, You can see the "qr" is already searched in search text field, which is retained over the screen orientations.

**Comment by PratyushSingh07** [comment]
> @PratyushSingh07, thanks for mentioning the bugs, Will make the search bar text smaller. Additionally for the second point that you have mentioned, Its not a bug, You can see the "qr" is already searched in search text field, which is retained over the screen orientations.

Ohhh I'm sorry, I had missed that

**Comment by AvneetSingh2001** [comment]
@therajanmaurya @PratyushSingh07 , I have made the required changes.

**Comment by therajanmaurya** [comment]
> @therajanmaurya Thanks for reviewing it, I will make the necessary changes.
> 
> @PratyushSingh07, thanks for mentioning the bugs, Will make the search bar text smaller. Additionally for the second point that you have mentioned, Its not a bug, You can see the "qr" is already searched in search text field, which is retained over the screen orientations.

Always remember follow material guidelines.

**Comment by PratyushSingh07** [comment]
@AvneetSingh2001 please squash your commits and follow our contribution guidelines . your commit message should have been 
`refactor #2447: migrated help screen to compose`

**Comment by AvneetSingh2001** [comment]
@PratyushSingh07  thanks for correcting my commit message, I have squashed the commits, have a look.

**Comment by PratyushSingh07** [comment]
Looks neat @AvneetSingh2001 . Great job !!

**Comment by AvneetSingh2001** [comment]
@PratyushSingh07 Should i also delete files that were used in XML view like adapter, they are now redundant as they are not being used anywhere else.

**Comment by PratyushSingh07** [comment]
> @PratyushSingh07 Should i also delete files that were used in XML view like adapter, they are now redundant as they are not being used anywhere else.

yes you can but be 100% sure before deleting them :) . And yes after this you will have to squash your commits again

**Comment by AvneetSingh2001** [comment]
@PratyushSingh07 done. Thanks a lot for your support :). Are we open to more compose migrations?

**Comment by therajanmaurya** [comment]
> @PratyushSingh07 done. Thanks a lot for your support :). Are we open to more compose migrations?

We want to migrate whole project in compose, You are welcome to send a PR. let me know when you are done with changes.

**Comment by AvneetSingh2001** [comment]
@therajanmaurya , the tasks of this PR is completed. 
Looking forward to more migrations.  Thanks :)

**Comment by therajanmaurya** [comment]
@AvneetSingh2001 can you please fix git conflicts ?

**Comment by AvneetSingh2001** [comment]
Hey @therajanmaurya , I have solved the conflicts.
Thanks.

**Comment by AvneetSingh2001** [comment]
@therajanmaurya   @PratyushSingh07  , i am not able to squash the commit after merging, is it okay or do i need to find some way to squash?

**Comment by PratyushSingh07** [comment]
This happens whenever you try to merge a branch into another branch. One way to fix this is by cherry picking. You can try it out if you want as it can be a life saver in some open source projects. But if you find it cumbersome then Rajan can do it at his end as well

**Comment by AvneetSingh2001** [comment]
ok, let me try the mentioned option then.
Thanks for your input @PratyushSingh07  :)

**Review by therajanmaurya — APPROVED** [review]

**Pull request merged**